### PR TITLE
CATFLOW Geospatial Tool

### PR DESCRIPTION
**Pull request by Ash-Manoj**
Restructured the run.py for generating required Raster files for CATFLOW Hillslope Wizard using a single Tool.

### CODE REVIEW

File: src/tool.yml [hunk 1]
```diff
@@ -8,88 +8,21 @@ tools:
...
+        default: false        
+      shapefile: 
         type: file
-        description: Input Raster filled DEM file 
-      z_factor: 
+        description: Input shapefile for clipping the processing extent           
+      stream_threshold: 
         type: float
-        description: Optional multiplier for when the vertical and horizontal units are not the same
-        default: None
```
Comment (mmaelicke) [inline src/tool.yml:34]: This parameter does not have a default, but is optional, as the tool will set it to 100. The difference between optional and default is a bit confusing. You can add `optional: true` here and add to the description that the threshold is always 100, but you can overwrite this constant

**Review by mmaelicke — CHANGES_REQUESTED** [review]
Nice work @Ash-Manoj ! Sorry for my late response, I forgot about this pull request. There are some minor changes requested.

**Comment by mmaelicke** [comment]
Any progress on this?

**Comment by Ash-Manoj** [comment]
> Any progress on this?

**Comment by Ash-Manoj** [comment]
Working on the changes. Should be able to push by next week.

File: src/run.py [hunk 2]
```diff
@@ -16,119 +16,67 @@
 if toolname == 'whitebox_info':
     wblib.print_info(to_file=kwargs.get('toFile', True))
 
- # Fill Depression tool
-elif toolname == 'fill_depressions':
+ # Tool for generating required Raster files for CATFLOW Hillslope Wizard
+elif toolname == 'catflow_hillslope_generator':
```
[the remark this answers is not in the document]
Comment (Ash-Manoj) [reply src/run.py:20]: Agreed. Changed to hillslope_generator.

File: src/run.py [hunk 3]
```diff
@@ -16,119 +16,67 @@
...
     # get the parameters
     try:
         inp = kwargs['dem']
-        out = '/out/filled_dem.tif'
-        flats = kwargs.get('fix_flats', True)
+        clip = kwargs.get('clip_extent', False)
+        shp = kwargs.get('shapefile', '/in/Shapefile.shp')
```
[the remark this answers is not in the document]
Comment (Ash-Manoj) [reply src/run.py:25]: Changed to Basin

File: src/run.py [hunk 4]
```diff
@@ -16,119 +16,67 @@
...
+        thres = kwargs.get('stream_threshold', 100.0)
```
[the remark this answers is not in the document]
Comment (Ash-Manoj) [reply src/run.py:26]: Discussion on default values for the flow accumulation.

File: src/tool.yml [hunk 5]
```diff
@@ -8,88 +8,21 @@ tools:
         type: bool
         description: If True, an INFO.txt will be written
         default: true
-  fill_depressions:
-    title: Fill Depressions
-    description: Fill Depressions in a DEM using WhiteboxGIS [fill_depression](https://www.whiteboxgeo.com/manual/wbt_book/available_tools/hydrological_analysis.html#FillDepressions) tool
+    catflow_hillslope_generator:
```
[the remark this answers is not in the document]
Comment (Ash-Manoj) [reply src/tool.yml:11]: Changed

**Pull request merged**